includes/class-boldgrid-inspirations-onboarding-progress.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * File: boldgrid-inspirations/includes/class-boldgrid-inspirations-onboarding-progress.php
+ *
+ * @package   Boldgrid_Inspirations_Onboarding_Progress
+ * @copyright BoldGrid.com
+ * @version   $Id$
+ * @author    BoldGrid.com <wpb@boldgrid.com>
+ */
+
+/**
+ * BoldGrid Inspirations Onboarding Progress class.
+ *
+ * This class is responsible for calculating the progress of
+ * the onboarding process. Additionally, this class is responsible
+ * updating the status of tasks as they are completed.
+ *
+ * @since 2.8.0
+ */
+class Boldgrid_Inspirations_Onboarding_Progress {
+	/**
+	 * The onboarding tasks option name.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var string
+	 */
+	var $tasks_option_name;
+
+	/**
+	 * The onboarding progress option name.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var string
+	 */
+	var $progress_option_name;
+
+	/**
+	 * Inspirations Configs
+	 *
+	 * @since 2.8.0
+	 *
+	 * @var array
+	 */
+	var $configs;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 2.8.0
+	 */
+	public function __construct( $configs ) {
+		$this->configs              = $configs;
+		$this->tasks_option_name    = $this->configs['onboarding_tasks_option'];
+		$this->progress_option_name = $this->configs['onboarding_progress_option'];
+	}
+
+	/**
+	 * Add Admin Ajax hooks.
+	 *
+	 * @since 2.8.0
+	 */
+	public function add_ajax_hooks() {
+		add_action( 'wp_ajax_boldgrid_inspirations_update_task', array( $this, 'ajax_update_task' ) );
+	}
+
+	/**
+	 * Get Task.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param string $task_id The task ID.
+	 *
+	 * @return array The task object.
+	 */
+	public function get_task( $task_id ) {
+		$tasks = get_option( $this->tasks_option_name );
```

Comment:
This set of code is duplicated several times.

```php
$tasks = get_option( $this->tasks_option_name );

if ( ! is_array( $tasks ) ) {
	return false;
}
```

What if we create a method get_tasks?

```php
function get_tasks() {
    $tasks = get_option( $this->tasks_option_name );

    return is_array( $tasks ) ? $tasks : array();
}
```